pkg/service/service.go

```diff
@@ -15,108 +15,172 @@ import (
 	"context"
 	"fmt"
 	"net/http"
+	"os"
 	"time"
 
 	"github.com/edgexfoundry/device-sdk-go/internal/autoevent"
+	"github.com/edgexfoundry/device-sdk-go/internal/clients"
 	"github.com/edgexfoundry/device-sdk-go/internal/common"
 	"github.com/edgexfoundry/device-sdk-go/internal/container"
 	"github.com/edgexfoundry/device-sdk-go/internal/controller"
 	dsModels "github.com/edgexfoundry/device-sdk-go/pkg/models"
+	bootstrapContainer "github.com/edgexfoundry/go-mod-bootstrap/bootstrap/container"
 	"github.com/edgexfoundry/go-mod-bootstrap/di"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+	"github.com/edgexfoundry/go-mod-registry/registry"
+	"github.com/gorilla/mux"
 
 	"github.com/edgexfoundry/go-mod-core-contracts/clients/types"
 	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
 	"github.com/google/uuid"
 )
 
 var (
-	svc *Service
+	sdk *DeviceServiceSDK
 )
 
-// A Service listens for requests and routes them to the right command
-type Service struct {
-	svcInfo     *common.ServiceInfo
-	asyncCh     chan *dsModels.AsyncValues
-	deviceCh    chan []dsModels.DiscoveredDevice
-	startTime   time.Time
-	controller  controller.RestController
-	initiazlied bool
+type DeviceServiceSDK struct {
+	ServiceName    string
+	LoggingClient  logger.LoggingClient
+	registryClient registry.Client
+	edgexClients   clients.EdgeXClient
+	controller     controller.RestController
+	config         *common.ConfigurationStruct
+	svcInfo        *common.ServiceInfo
```

Comment:
This is contained in `ConfigurationStruct`, so why is it need here when can get to it via `config`?

Reply:
you're right, updated.